### DISCUSSION TITLE
Alpine js plugin doesn't work

### DISCUSSION THREAD

**Discussion by memiljamel** [alpinejs/alpine#3066] [1. Help]
Hi, why when I declare defer in alpinejs script tag, the mask plugin function doesn't work.

Here I am using laravel.

// resources/js/app.js
```
import Alpine from 'alpinejs';
import mask from '@alpinejs/mask';

alpine.plugin(mask);

window.Alpine = Alpine;

alpine.start();
```

// resources/view/welcome.blade.php
```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>{{ config('app.name', 'Laravel') }}</title>
  <link rel="stylesheet" href="{{ mix('css/app.css') }}" />
  <script src="{{ mix('js/manifest.js') }}"></script>
  <script src="{{ mix('js/vendor.js') }}" defer></script>
  <script src="{{ mix('js/app.js') }}"></script>
</head>
<body>

<input x-mask="99/99/9999" placeholder="MM/DD/YYYY" />

</body>
</html>
```

when I remove the defer declaration from the script tag it works, but there is an error in the console log Alpine Warning: Unable to initialize. Trying to load Alpine before `<body>` is available. Did you forget to add `defer` in Alpine's `<script>` tag?

**Comment by SimoTod** [comment]
It does work with defer. You didn't have any alpine components in your page (no x-data).

**Comment by joaopalopes24** [comment]
`x-data` is missing.

**Comment by DaviMenezes** [comment]
`<input x-data x-mask="99/99/9999">` x-data missing ... 

> they should include something about it in the official documentation, if it already exists I don't know where it is, but I already had a lot of trouble finding this information

**Reply by ekwoka** [reply]
https://alpinejs.dev/start-here#declaring-data

It's about the first thing the docs cover.

**Reply by boosh** [reply]
Having lost the afternoon to this, this should be emphasised in bold on the "Extending" page as well as on plugin pages :(
